starknet_py/tests/e2e/account/account_test.py

```diff
@@ -179,6 +180,30 @@ async def test_sign_declare_transaction(gateway_account, map_compiled_contract):
         map_compiled_contract, max_fee=MAX_FEE
     )
 
+    assert isinstance(signed_tx, Declare)
+    assert signed_tx.version == 1
+    assert isinstance(signed_tx.signature, list)
+    assert len(signed_tx.signature) > 0
+    assert signed_tx.max_fee == MAX_FEE
+
+
+@pytest.mark.asyncio
+async def test_sign_declare_v2_transaction(
+    gateway_account, sierra_minimal_compiled_contract_and_class_hash
+):
+    (
+        compiled_contract,
+        compiled_class_hash,
+    ) = sierra_minimal_compiled_contract_and_class_hash
+
+    signed_tx = await gateway_account.sign_declare_transaction(
+        compiled_contract=compiled_contract,
+        compiled_class_hash=compiled_class_hash,
+        max_fee=MAX_FEE,
```

Comment:
I suppose `auto_estimate` won't work until #842 is resolved?
It should be tested here.

Reply:
Wish we could test any kind of fee estimation, but it is not working atm. #842 will be resolved (and tested hopefully) in the separate PR.

Reply:
We should do something with this thread, not to forget to test `auto_estimate` here